gnat2goto/driver/tree_walk.adb

```diff
@@ -1480,7 +1481,14 @@ package body Tree_Walk is
       Typecast_Expr : constant Irep :=
         Make_Op_Typecast (Value, Sloc (N), Type_Of_Val);
    begin
-      return Make_Range_Assert_Expr (N, Typecast_Expr, Type_Of_Val);
+      --  TODO: Range expressions for non-bounded types are outside
+      --        the scope of this PR
+      if Kind (Type_Of_Val) in I_Bounded_Signedbv_Type | I_Bounded_Floatbv_Type
+      then
+         return Make_Range_Assert_Expr (N, Typecast_Expr, Type_Of_Val);
+      else
+         return Typecast_Expr;
```

Comment:
comment implies that we should be emitting a range check here but don't for implementation reasons... I don't think we should have a diagnostic for that because that'd get extremely annoying very quickly, but we should mark what needs to be done here in some way